Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,253 @@
+        machine(str): the machine to lock.
+        machines_locks(srt): all the exiting lock files.
+        gitlab_status_token(str): the gitlab token.
+        gcs_locks_path(str): the lock repository name.
+        job_id(str): the job id to check.
+
+    Returns: the machine name if locked.
+
+    """
+    lock_machine_name = ''
+    job_id_of_the_existing_lock = next((d['job_id'] for d in machines_locks if d["machine_name"] == machine), None)
+    if job_id_of_the_existing_lock:
+        logging.info(f'There is a lock file for job id: {job_id_of_the_existing_lock}')
+        job_id_of_the_existing_lock_status = check_job_status(gitlab_status_token, job_id_of_the_existing_lock)
+        logging.info(f'the status of job id: {job_id_of_the_existing_lock} is: {job_id_of_the_existing_lock_status}')
+        if job_id_of_the_existing_lock_status != 'running':
+            # machine found! removing the not relevant lock and create a now one
+            remove_machine_lock_file(storage_bucket, gcs_locks_path, machine, job_id_of_the_existing_lock)
+            lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
+            lock_machine_name = machine
+    else:
+        # machine found! create lock file
+        logging.info('There is no a lock file')
+        lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
+        lock_machine_name = machine
+    return lock_machine_name
+
+
+def main():
+    install_logging('lock_cloud_machines.log', logger=logging)
+    logging.info('Starting search a CLOUD machine to lock')
+    options = options_handler()
+    storage_client = storage.Client.from_service_account_json(options.service_account)
+    storage_bucket = storage_client.bucket('xsoar-ci-artifacts')
+    number_machines_to_lock = int(options.number_machines_to_lock)
+
+    logging.info('adding job_id to the queue')
+    adding_build_to_the_queue(storage_bucket, options.gcs_locks_path, options.ci_job_id)
+
+    # running until the build is the first in queue
+    first_in_the_queue = False
+    while not first_in_the_queue:
+        my_place_in_the_queue, previous_build = get_my_place_in_the_queue(storage_client, options.gcs_locks_path,
+                                                                          options.ci_job_id)
+        logging.info(f'my place in the queue is: {my_place_in_the_queue}')
+
+        if my_place_in_the_queue == 0:
+            first_in_the_queue = True
+        else:
+            # we check the status of the build that is ahead of me in the queue
+            previous_build_status = check_job_status(options.gitlab_status_token, previous_build)
+            if previous_build_status != 'running':
+                # delete the lock file of the build because its not running
+                remove_build_from_queue(storage_bucket, options.gcs_locks_path, previous_build)
+            else:
+                sleep(random.randint(8, 13))
```

Comment:
I think this entire code block should be extracted to an additional function `wait_for_build_to_be_first_in_queue()`